### PR TITLE
[feat] Adding camvid segmentation multiclass as an example

### PR DESCRIPTION
**Pull request by TimbusCalin**
- Multiclass semantic segmentation example on all classes
- Background class is added as an additional class
- Initial PyTorch Lightning example is maintained, adapted in accordance with multiclass

### CODE REVIEW

**Comment by qubvel** [comment]
Hi @TimbusCalin, thanks a lot for your contribution! Looks great overall and it's pretty close to being merged, here are just a few minor comments:

1. Remove commented code or replace it with comments / assertations
```
# Specify classes only if necessary
# CLASSES = ['road','car']
```
```
# Debugging unique values and type of mask
# print(f"Mask dtype after casting: {mask.dtype}, unique values: {torch.unique(mask)}, shape: {mask.shape}")
```
```
# Debugging logits shape
# print(f"Logits mask shape: {logits_mask.shape}")
```
maybe somewhere else

2. It would also be cool to add metric computation per class, if you have time and know how to do that :) but that's optional

**Comment by abdksyed** [comment]
I think the OUT_CLASSES should be equal to len of CLASSES. Since the background(unlabelled) classes is already included in the CLASSES list.

```python
OUT_CLASSES = len(train_dataset.CLASSES) + 1
```

This should be 
```python
OUT_CLASSES = len(train_dataset.CLASSES)
```

**Comment by abdksyed** [comment]
> Hello @abdksyed @qubvel.
> 
> I am fine with this approach that @abdksyed mentioned, to integrate my PR now, and I will let @abdksyed do the class-metrics-wise function now, once the PR is integrated; glad that we synchronized on this.
> 
> @abdksyed You are right, that's why I updated it accordingly in this commit here [aa8226c](https://github.com/qubvel-org/segmentation_models.pytorch/pull/922/commits/aa8226c05df1d83222b491b1a822a0c212421075)

Oh sorry, I missed your last commit. Apologies

**Comment by TimbusCalin** [comment]
@abdksyed no issue, another pair of (good!) eyes is always necessary and welcome!

**Comment by CalinLucian** [comment]
@qubvel I guess the dilemma is solved now. We may integrate this, if you see it fit.

**Comment by TimbusCalin** [comment]
@qubvel I guess the dillema is solved. We can integrate if you have no other remarks.

**Comment by qubvel** [comment]
Thanks for your contribution!

**Comment by sajeedmehrab** [comment]
@abdksyed Do you have an example for how we can compute the class-wise miou?

**Comment by abdksyed** [comment]
@sajeedmehrab 
I use the [torchmetrics](https://lightning.ai/docs/torchmetrics/stable/segmentation/mean_iou.html). You can give the `per_class` as True to get class wise iou.

**Comment by qubvel** [comment]
Would appreciate if you contribute torch metrics example 😊
